src/dogroups.c

```diff
@@ -48,8 +48,8 @@ SEXP dogroups(SEXP dt, SEXP dtcols, SEXP groups, SEXP grpcols, SEXP jiscols, SEX
   iSD = PROTECT(findVar(install(".iSD"), env)); protecti++; // 1-row and possibly no cols (if no i variables are used via JIS)
   xSD = PROTECT(findVar(install(".xSD"), env)); protecti++;
   R_len_t maxGrpSize = 0;
-  const int *ilens = INTEGER(lens);
-  for (R_len_t i=0; i<LENGTH(lens); i++) {
+  const int *ilens = INTEGER(lens), n=LENGTH(lens);
+  for (R_len_t i=0; i<n; ++i) {
```

Comment:
@mattdowle in this commit many changes from `i++` to `++i` or reverse, but at the same time no logic change, what is the reason for that?

Reply:
Did you see https://github.com/Rdatatable/data.table/pull/3610#issuecomment-497099447?

Reply:
found answer in PR comment